execution/executor/src/chunk_executor.rs

```diff
@@ -313,6 +321,60 @@ impl<V: VMExecutor> TransactionReplayer for ChunkExecutor<V> {
 
 impl<V: VMExecutor> TransactionReplayer for ChunkExecutorInner<V> {
     fn replay(
+        &self,
+        mut transactions: Vec<Transaction>,
+        mut transaction_infos: Vec<TransactionInfo>,
+        writesets: Vec<WriteSet>,
+        events: Vec<Vec<ContractEvent>>,
+        txns_to_skip: Arc<BTreeSet<Version>>,
+    ) -> Result<()> {
+        let current_begin_version = {
+            self.commit_queue
+                .lock()
+                .persisted_and_latest_view()
+                .1
+                .version()
```

Comment:
this is None for the chunk containing the genesis. Needs something like this https://github.com/aptos-labs/aptos-core/pull/5848